src/Inventory.zig

```diff
@@ -505,71 +518,187 @@ pub const Command = struct { // MARK: Command
 			durability: u31,
 			previousDurability: u32 = undefined,
 		},
+		addHealth: struct {
+			target: ?*main.server.User,
+			health: f32,
+			cause: main.game.DamageType,
+			previous: f32
+		}
+	};
+
+	const SyncOperationType = enum(u8) {
+		create = 0,
+		delete = 1,
+		useDurability = 2,
+		health = 3
 	};
 
-	const SyncOperation = struct { // MARK: SyncOperation
+	const SyncOperation = union(SyncOperationType) { // MARK: SyncOperation
 		// Since the client doesn't know about all inventories, we can only use create(+amount)/delete(-amount) and use durability operations to apply the server side updates.
-		inv: InventoryAndSlot,
-		amount: i32,
-		item: ?Item,
-		durability: i32,
+		create: struct {
+			inv: InventoryAndSlot,
+			amount: u16,
+			item: ?Item
+		},
+		delete: struct {
+			inv: InventoryAndSlot,
+			amount: u16
+		},
+		useDurability: struct {
+			inv: InventoryAndSlot,
+			durability: u32
```

Comment:
Just to clarify:
You changed it to unsigned because you think there should be a separate entry to add durability, if that is going to be a thing in the future?